example/extensions/lib_subgraph/subgraph_lib.cc

```diff
@@ -218,10 +218,27 @@ MXReturnValue mySupportedOps(std::string json,
 
 MXReturnValue myReviewSubgraph(std::string json, int subraph_id, bool* accept,
                                std::unordered_map<std::string, std::string>& options,
-                               std::unordered_map<std::string, std::string>& attrs) {
+                               std::unordered_map<std::string, std::string>& attrs,
+                               std::map<std::string, MXTensor>& args,
+                               std::map<std::string, MXTensor>& aux) {
   for (auto kv : options) {
     std::cout << "option: " << kv.first << " ==> " << kv.second << std::endl;
   }
+  for (auto kv : args) {
+    std::cout << "arg: " << kv.first << " ==> (";
+    for (auto s : kv.second.shape)
+      std::cout << s << ",";
+    std::cout << ") [";
+    for (int i=0; i<kv.second.size(); i++)
+      std::cout << kv.second.data<float>()[i] << ", ";
+    std::cout << "]" << std::endl;
+  }
+  if(options.count("reqArgs") > 0 && args.size() == 0) {
```

Comment:
ditto missing auxs again ?
we should probably have a reqAuxs option too

Reply:
this is just an example implementation of the subgraph library. it can do whatever the user wants it to do. 

Theres no reason to require aux by default, depending on the model it may or may not have any aux params. The smarter thing a user could do is check and see if the subgraph has aux inputs, and then in the aux map is empty error out.

Reply:
Agreed